mule-extensions-api/src/main/java/org/mule/runtime/extension/api/options/OptionsResolver.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.options;
+
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.toSet;
+import org.mule.runtime.api.options.Option;
+import org.mule.runtime.api.options.OptionBuilder;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * mule-runtime-project
+ *
+ * @author Esteban Wasinger (http://github.com/estebanwasinger)
+ */
+public interface OptionsResolver {
+
+  Set<Option> resolve();
+
+  default Set<Option> getOptionsFor(String... values) {
```

Comment:
don't put this util methods as part of the resolver interface. An OptionsResolver should never override this methods. Add this "simplified" builders as part of the OptionBuilder